src/errors.ts

```diff
@@ -112,3 +116,43 @@ export enum ValidationErrorCodes {
     InternalError = 1008,
     TokenNotSupported = 1009,
 }
+
+export abstract class AlertError {
+    public abstract message: string;
+    public shouldAlert: boolean = true;
+}
+
+export class ExpiredOrderError extends AlertError {
+    public message = `Found expired order! `;
+    public expiry: number;
+    public expiredForSeconds: number;
+    constructor(public order: SignedOrder, public currentThreshold: number, public details?: string) {
+        super();
+        this.expiry = order.expirationTimeSeconds.toNumber();
+        this.expiredForSeconds = Date.now() / ONE_SECOND_MS - this.expiry;
+    }
+}
+
+export class OrderWatcherSyncError extends AlertError {
+    public message = `Error syncing OrderWatcher!`;
+    constructor(public details?: string) {
+        super();
+    }
+}
+
+/**
+ * If the max age of expired orders exceeds the configured threshold, this function
+ * logs an error capturing the details of the expired orders
+ */
+export function alertOnExpiredOrders(expired: APIOrder[], details?: string): void {
+    if (expired.length > 0) {
+        const descSort = expired.sort((a, b) =>
+            b.order.expirationTimeSeconds.minus(a.order.expirationTimeSeconds).toNumber(),
+        );
+        const maxExpiration = descSort[0].order.expirationTimeSeconds;
+        if (maxExpiration > MAX_EXPIRATION_ALERT_THRESHOLD_SECONDS) {
+            const error = new ExpiredOrderError(descSort[0].order, MAX_EXPIRATION_ALERT_THRESHOLD_SECONDS, details);
+            logger.error(error);
```

Comment:
If we will use the default object serialiser, we might include unnecessary fields (such as `shouldAlert` , just something to be mindful of :)

Reply:
I was thinking the default serialiser will include all fields in the object, is that right?

Reply:
Yes